sway-core/src/asm_generation/from_ir.rs

```diff
@@ -2639,7 +2639,7 @@ mod tests {
 
         let asm_script = format!("{}", asm);
         if asm_script != expected {
-            print!("{}", prettydiff::diff_lines(&expected, &asm_script));
+            tracing::info!("{}", prettydiff::diff_lines(&expected, &asm_script));
```

Comment:
This might not be *quite* the same - it looks like `print!` was used previously which (unlike `println`) doesn't insert a newline at the end, whereas `info!` does. That said, I don't think it matters as this is debug output? cc @otrho

Reply:
True! My guess was that it didn't matter since this looks like it's printing a large blob, and the newline would only be applied at the end. 

A few cases I've seen where it's important to preserve `print` without a newline are where we're expecting user input on the same line, such as `print!("Please provide the signature for this transaction:")`

Do you think should should be using `tracing::debug` instead? (That is, only visible with `--verbose`)

Reply:
I think it's fine to use `info!` for now, as I believe printing this is already gated behind another compiler flag - it might be too confusing to have to enable two different flags.

Reply:
Ahh on a closer look, it seems it's only printed in the test anyway! It's probably fine if we close this and let the test just use `print!`? Using `tracing` is more about giving the user a consistent form of control over the output of forc and related tools, and these tests aren't exposed publicly to the user anyway.